src/kakarot/instructions/system_operations.cairo

```diff
@@ -102,7 +102,7 @@ namespace SystemOperations {
 
         // Check sender balance and nonce
         let sender = State.get_account(evm.message.address.evm);
-        let is_nonce_overflow = is_le(Constants.MAX_NONCE + 1, sender.nonce);
+        let is_nonce_overflow = Helpers.is_zero(Constants.MAX_NONCE - sender.nonce);
```

Comment:
can there be a case where sender.nonce > Constants.MAX_NONCE and this substraction wraps around?